python/unitytrainers/self_play_trainer_controller.py

```diff
@@ -16,8 +18,13 @@ def __init__(self, env_path, run_id, save_freq, curriculum_folder, fast_simulati
         self.ghost_trainers_path = self.model_path + 'self-play-checkpoints/'
         if not os.path.exists(self.ghost_trainers_path):
             os.makedirs(self.ghost_trainers_path)
-        raise UnityTrainerControllerException("The handle_episode_termination method was not implemented as hyper parameters need to be included")
-        # TODO add opci, delta and opponent save frequency hyperparameters
+
+        self.delta = None
+        self.opponent_policy_change_interval = None
+        self.ghost_save_frequency = None
+
+        self.elapsed_episodes = 0
+        raise UnityTrainerControllerException("The __init__ method was not implemented.")
```

Comment:
What aspect is yet to be implemented?

Reply:
It might be better to specify what exactly needs to be implemented.

Reply:
What is left to be implemented is to specify where the self play hyper parameters are introduced. Most likely in the `trainer_config.yaml`?